### DISCUSSION TITLE
DOS4GW 2.01

### DISCUSSION THREAD

**Discussion by javiergutierrezchamorro** [open-watcom/open-watcom-v2#193] [General]
I have achieved to get the permission from Terry Colligan at Tenberry, to redistribute latest dos4gw 2.01 into OpenWatcom, as an upgrade of current 1.97: http://www.tenberry.com/dos4g/watcom/rn4gw.html

It was a paid upgrade, that he is kindly providing to the project, on a royalte free basis.

Unfortunately his backup server got broken, and he will need about a month to recover it. In the meantime, we can upgrade dos4gw.exe in the repository, or either wait for the system restore when it happens, and then upgrade all the distribution files related to DOS4GW.

Is it interested for you? If so, I can handle it. Do you prefer to upgrade the exe now, and later related files, or do it at the same time?

Thanks.

**Comment by jmalak** [comment]
It is excellent, thanks for your effort.
We should do some testing with this new version of DOS4GW before we include it in OW 2.0 to be sure that no problem with compatibility. By example sampler, debugger, executable(stub) etc.

**Comment by javiergutierrezchamorro** [comment]
http://www.javiergutierrezchamorro.com/_temp/upload/DOS4GW%202.01a%20(EXE%20only).zip

**Comment by jmalak** [comment]
Thanks for link.

**Comment by javiergutierrezchamorro** [comment]
Tried creating a commit before, but could not with a pending pull request.

**Comment by jmalak** [comment]
You can, but you must create another branch.

**Comment by Azarien** [comment]
DOS4GW 2.01a does not work with set DOS4G=QUIET

**Comment by jmalak** [comment]
What you mean by "does not work with set DOS4G=QUIET"?
Do you get banner, but application is running or not?

**Comment by javiergutierrezchamorro** [comment]
Assuming DOS4G=QUIET works on 1.97, and not in 2.01a, and assuming the the documentation does not specify so, probably better thing would be report it to Tenberry.

At the moment DOW4GW 2.01a is only a binary distribution in OW. I am waiting for Terry for the full release yet.

**Comment by Azarien** [comment]
```
c:\>dos4gw
DOS/4GW Protected Mode Run-time  Version 2.01a
Copyright (c) Tenberry Software, Inc. 1996
DOS/4GW fatal error (1004): syntax is DOS4GW <executable.xxx>

c:\>set dos4g=quiet

c:\>dos4gw
DOS/4G error (5004): cfg syntax error - Invalid char after option string
DOS4G=quiet
           ^
```

This prevents an application from running.
I haven't tested other variables.
v 1.97 works fine.

**Comment by javiergutierrezchamorro** [comment]
Agree Azarien.
Have been able to reproduce it both under DR-DOS 8 and under DOSBOX.
Hope Tenberry will answer your report.

**Comment by igorzhitko** [comment]
Try set DOS4GW=STARTUPBANNER:OFF

**Comment by Azarien** [comment]
Doesn't seem to do anything (neither with DOS4G)

Maybe QUIET is simply not supported in 2.01. But the message is quite confusing.

**Comment by jmalak** [comment]
In release notes is following section

E *\* Configuration has been enhanced for flexibility and consistency
     across the DOS/4G Family of products. Most of the options have
     remained the same.  Please review chapter 4 of the the DOS/4G Family
     Programmer's Guide for details.
     Note: The new format for configuration options is not backwards
     compatible with the 1.97 version configuration options. You must
     convert your preexisting options to the new format or they will
     not work.

It is problem with OW stub which use old parameter QUIET.
But OW 2.0 use stub without QUIET parameter that OW 2.0 compilers should work with DOS4GW 2.0x.

Please, could you check OW2.0 DOS compilers if they works with new DOS4GW.

**Comment by jmalak** [comment]
I added note to OW wiki about DOS4GW configuration to work properly with OW 2.0.
https://github.com/open-watcom/open-watcom-v2/wiki/Notes

**Comment by Azarien** [comment]
Is the documentation available online somewhere?

**Comment by javiergutierrezchamorro** [comment]
We are waiting for Terry for the full release. At the moment, only EXE is legally entitled.

**Comment by javiergutierrezchamorro** [comment]
Terry Colligan is facing some health issues, and have not been able to recover a full copy of DOS/4GW.
Anyway, he remember his authorization to include newer DOS/4GW.

Unfortunately have not been able to get full package from other sources yet. If any one has it, please attach it here so I can push to the repository.

In the meantime @jmalak can you please update bld/redist with this EXE only?
[dos4gw 2.01a.zip](https://github.com/open-watcom/open-watcom-v2/files/1298864/dos4gw.2.01a.zip)

Thanks.

**Comment by javiergutierrezchamorro** [comment]
Unfortunately Terry passed away a few months back:
- http://www.javiergutierrezchamorro.com/dep-terry-colligan/3678
- http://www.bttr-software.de/forum/forum_entry.php?id=15516

Hope someone will have a full 2.01a copy to be integrated in Openwatcom. Will be a nice gift for remembering him.

**Reply by NiQ1** [reply]
> Unfortunately Terry passed away a few months back:
> 
> * http://www.javiergutierrezchamorro.com/dep-terry-colligan/3678
> * http://www.bttr-software.de/forum/forum_entry.php?id=15516
> 
> Hope someone will have a full 2.01a copy to be integrated in Openwatcom. Will be a nice gift for remembering him.

Been many years but here's a link -
https://vetusware.com/download/Dos4GW%20Professional%202.01a%202.01a/?id=6239
Download requires registration but I've downloaded it and it seems legit.

**Comment by javiergutierrezchamorro** [comment]
What about DOS/4G Professional 2.60?
https://ia802701.us.archive.org/zipview.php?zip=/34/items/scratchupload/scratchupload.zip

**Comment by AnthraxBass** [comment]
> What about DOS/4G Professional 2.60?
> https://ia802701.us.archive.org/zipview.php?zip=/34/items/scratchupload/scratchupload.zip

Unfortunately, both versions (dos4gw & dos4g) from the given archive ([DOS Application] Tenberry - DOS4G 2.60 Professional - Distribution Disks [http---usuarios.lycos.es-scratchupload].zip) cannot handle "quiet" option.

**Comment by jmalak** [comment]
You can fix it by setup described here
https://github.com/open-watcom/open-watcom-v2/wiki/Notes

**Comment by eoredson** [comment]
When starting with an .exe in Dosbox-X all I get is:

DOS/4GW fatal error (1008): can't load executable format for file DNDBBS.EXE [1]

**Comment by AnthraxBass** [comment]
> When starting with an .exe in Dosbox-X all I get is:
> 
> DOS/4GW fatal error (1008): can't load executable format for file DNDBBS.EXE [1]

Try some of the suggestions from this forum: https://misterfpga.org/viewtopic.php?t=1663&start=30

**Comment by bret-walda** [comment]
what is the difference between 2.01 and 2.60 and which one should one use?

**Reply by javiergutierrezchamorro** [reply]
> what is the difference between 2.01 and 2.60 and which one should one use?

2.60 is the professional version, while 2.01 is the one that come with Watcom, upgraded from 1.97 to 2.01.

You can read more here:
- https://web.archive.org/web/20150809183503/http://www.tenberry.com/dos4g/watcom/index.html
- https://web.archive.org/web/20150601002504/http://tenberry.com/dos4g/index.html
